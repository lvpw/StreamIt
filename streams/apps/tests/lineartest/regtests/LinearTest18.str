/** 
 * Test out combining splitjoins, with multiple split 
 * joins in weighted roundrobin fashion.
 **/
void->void pipeline LinearTest18 {
  add FloatSource();
  add LinearSJ();
  add FloatSink();
}

void->float stateful filter FloatSource {
  int x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = (x + 1)%20;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/** A linear split join to test combination with different round robin weightings. **/
float->float splitjoin LinearSJ {
  split duplicate;
  for (int i=0; i<10; i++) {
    add MockFIRFilter(i);
  }
  join roundrobin;

}

/** This is a mock up of an FIR filter. N is where the counting starts from **/
float->float filter MockFIRFilter(int N) {
  float[10] weights;
  init {
    for (int i=0; i<10; i++) {
      weights[i] = N + i;
    }
  }
  work pop 1 peek 10 push 1 {
    float sum = 0;
    /** do the MAC like normal **/
    for (int i=0; i<10; i++) {
      sum = sum + (peek(i) * weights[i]);
    }
    push(sum);
    pop();
  }
}
