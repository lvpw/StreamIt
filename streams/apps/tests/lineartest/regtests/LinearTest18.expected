LinearSJ_1.matrix
[[18.0+0.0i 17.0+0.0i 16.0+0.0i 15.0+0.0i 14.0+0.0i 13.0+0.0i 12.0+0.0i 11.0+0.0i 10.0+0.0i 9.0+0.0i]
 [17.0+0.0i 16.0+0.0i 15.0+0.0i 14.0+0.0i 13.0+0.0i 12.0+0.0i 11.0+0.0i 10.0+0.0i 9.0+0.0i 8.0+0.0i]
 [16.0+0.0i 15.0+0.0i 14.0+0.0i 13.0+0.0i 12.0+0.0i 11.0+0.0i 10.0+0.0i 9.0+0.0i 8.0+0.0i 7.0+0.0i]
 [15.0+0.0i 14.0+0.0i 13.0+0.0i 12.0+0.0i 11.0+0.0i 10.0+0.0i 9.0+0.0i 8.0+0.0i 7.0+0.0i 6.0+0.0i]
 [14.0+0.0i 13.0+0.0i 12.0+0.0i 11.0+0.0i 10.0+0.0i 9.0+0.0i 8.0+0.0i 7.0+0.0i 6.0+0.0i 5.0+0.0i]
 [13.0+0.0i 12.0+0.0i 11.0+0.0i 10.0+0.0i 9.0+0.0i 8.0+0.0i 7.0+0.0i 6.0+0.0i 5.0+0.0i 4.0+0.0i]
 [12.0+0.0i 11.0+0.0i 10.0+0.0i 9.0+0.0i 8.0+0.0i 7.0+0.0i 6.0+0.0i 5.0+0.0i 4.0+0.0i 3.0+0.0i]
 [11.0+0.0i 10.0+0.0i 9.0+0.0i 8.0+0.0i 7.0+0.0i 6.0+0.0i 5.0+0.0i 4.0+0.0i 3.0+0.0i 2.0+0.0i]
 [10.0+0.0i 9.0+0.0i 8.0+0.0i 7.0+0.0i 6.0+0.0i 5.0+0.0i 4.0+0.0i 3.0+0.0i 2.0+0.0i 1.0+0.0i]
 [9.0+0.0i 8.0+0.0i 7.0+0.0i 6.0+0.0i 5.0+0.0i 4.0+0.0i 3.0+0.0i 2.0+0.0i 1.0+0.0i 0.0+0.0i]]
LinearSJ_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
MockFIRFilter_1.matrix
[[9.0+0.0i]
 [8.0+0.0i]
 [7.0+0.0i]
 [6.0+0.0i]
 [5.0+0.0i]
 [4.0+0.0i]
 [3.0+0.0i]
 [2.0+0.0i]
 [1.0+0.0i]
 [0.0+0.0i]]
MockFIRFilter_1.vector
[[0.0+0.0i]]
MockFIRFilter_10.matrix
[[18.0+0.0i]
 [17.0+0.0i]
 [16.0+0.0i]
 [15.0+0.0i]
 [14.0+0.0i]
 [13.0+0.0i]
 [12.0+0.0i]
 [11.0+0.0i]
 [10.0+0.0i]
 [9.0+0.0i]]
MockFIRFilter_10.vector
[[0.0+0.0i]]
MockFIRFilter_2.matrix
[[10.0+0.0i]
 [9.0+0.0i]
 [8.0+0.0i]
 [7.0+0.0i]
 [6.0+0.0i]
 [5.0+0.0i]
 [4.0+0.0i]
 [3.0+0.0i]
 [2.0+0.0i]
 [1.0+0.0i]]
MockFIRFilter_2.vector
[[0.0+0.0i]]
MockFIRFilter_3.matrix
[[11.0+0.0i]
 [10.0+0.0i]
 [9.0+0.0i]
 [8.0+0.0i]
 [7.0+0.0i]
 [6.0+0.0i]
 [5.0+0.0i]
 [4.0+0.0i]
 [3.0+0.0i]
 [2.0+0.0i]]
MockFIRFilter_3.vector
[[0.0+0.0i]]
MockFIRFilter_4.matrix
[[12.0+0.0i]
 [11.0+0.0i]
 [10.0+0.0i]
 [9.0+0.0i]
 [8.0+0.0i]
 [7.0+0.0i]
 [6.0+0.0i]
 [5.0+0.0i]
 [4.0+0.0i]
 [3.0+0.0i]]
MockFIRFilter_4.vector
[[0.0+0.0i]]
MockFIRFilter_5.matrix
[[13.0+0.0i]
 [12.0+0.0i]
 [11.0+0.0i]
 [10.0+0.0i]
 [9.0+0.0i]
 [8.0+0.0i]
 [7.0+0.0i]
 [6.0+0.0i]
 [5.0+0.0i]
 [4.0+0.0i]]
MockFIRFilter_5.vector
[[0.0+0.0i]]
MockFIRFilter_6.matrix
[[14.0+0.0i]
 [13.0+0.0i]
 [12.0+0.0i]
 [11.0+0.0i]
 [10.0+0.0i]
 [9.0+0.0i]
 [8.0+0.0i]
 [7.0+0.0i]
 [6.0+0.0i]
 [5.0+0.0i]]
MockFIRFilter_6.vector
[[0.0+0.0i]]
MockFIRFilter_7.matrix
[[15.0+0.0i]
 [14.0+0.0i]
 [13.0+0.0i]
 [12.0+0.0i]
 [11.0+0.0i]
 [10.0+0.0i]
 [9.0+0.0i]
 [8.0+0.0i]
 [7.0+0.0i]
 [6.0+0.0i]]
MockFIRFilter_7.vector
[[0.0+0.0i]]
MockFIRFilter_8.matrix
[[16.0+0.0i]
 [15.0+0.0i]
 [14.0+0.0i]
 [13.0+0.0i]
 [12.0+0.0i]
 [11.0+0.0i]
 [10.0+0.0i]
 [9.0+0.0i]
 [8.0+0.0i]
 [7.0+0.0i]]
MockFIRFilter_8.vector
[[0.0+0.0i]]
MockFIRFilter_9.matrix
[[17.0+0.0i]
 [16.0+0.0i]
 [15.0+0.0i]
 [14.0+0.0i]
 [13.0+0.0i]
 [12.0+0.0i]
 [11.0+0.0i]
 [10.0+0.0i]
 [9.0+0.0i]
 [8.0+0.0i]]
MockFIRFilter_9.vector
[[0.0+0.0i]]
